src/PaletteProvider.js

```diff
@@ -144,35 +126,47 @@ export default class MaterialPaletteProvider
     if (this.props.defaults) {
       validateDefaults(this.props.defaults);
     }
-    execIfFunction(this.props.onInit);
-    createMaterialPalette(this.props.image, this.props.options)
-      .then((palette: PaletteInstance) => {
+
+    this._createPalette();
+  }
+
+  _createPalette = () => {
+    const { image, options, onFinish, onError } = this.props;
+
+    createMaterialPalette(image, options).then(
+      palette => {
         const paletteWithDefaults = this._mergeWithDefaults(palette);
-        execIfFunction(this.props.onFinish, paletteWithDefaults);
-        if (!this.props.forceRender) {
-          this.setState({ palette: paletteWithDefaults });
-        }
+
+        if (onFinish) onFinish(paletteWithDefaults);
+
+        this.setState({ palette: paletteWithDefaults });
         this.eventEmitter.publish({
```

Comment:
We need to `publish` before setting state, remember the issue we discussed at the office. The change is in my branch anyway, but will be good to have it here as well already 🤖